x/staking/keeper/invariants.go

```diff
@@ -112,42 +108,42 @@ func NonNegativePowerInvariant(k Keeper) sdk.Invariant {
 			powerKey := types.GetValidatorsByPowerIndexKey(validator)
 
 			if !bytes.Equal(iterator.Key(), powerKey) {
-				return fmt.Errorf("power store invariance:\n\tvalidator.Power: %v"+
+				return fmt.Sprintf("power store invariance:\n\tvalidator.Power: %v"+
 					"\n\tkey should be: %v\n\tkey in store: %v",
-					validator.GetConsensusPower(), powerKey, iterator.Key())
+					validator.GetConsensusPower(), powerKey, iterator.Key()), true
 			}
 
 			if validator.Tokens.IsNegative() {
-				return fmt.Errorf("negative tokens for validator: %v", validator)
+				return fmt.Sprintf("negative tokens for validator: %v", validator), true
```

Comment:
Again think we should accumulate all negative validators and return string with all errors displayed